lib/service_provider.rb

```diff
@@ -9,14 +9,18 @@ def initialize(host)
   end
 
   def metadata
-    ignored_methods = [:metadata, :encrypt_responses?, :encryption_opts]
+    ignored_methods = [:metadata, :encrypt_responses?, :encryption_opts, :issuer]
     metadata_methods = self.class.instance_methods(false) - ignored_methods
```

Comment:
agreed - let's save that. FWIW, I think the blacklist is still the right choice.